src/service/processing.py

```diff
@@ -256,3 +256,69 @@ async def heartbeat(millis: int):
             hb.stop()
         if arangoclient:
             await arangoclient.close()
+
+
+class SubsetSpecification:
+    """
+    An ID and associated information for a subset (either a match or selection).
+    
+    Instance variables:
+    internal_subset_id - the ID from the internal_subset_id argument or the
+        completed subset process, or None if not provided.
+    mark_only - whether then subset should be filtered out (False) or just marked (True).
+    prefix - the prefix to be applied to the subset ID.
+    """
+    
+    def __init__(
+            self,
+            internal_subset_id: str = None,
+            subset_process: models.DataProductProcess = None,
+            mark_only: bool = False,
+            prefix: str = None,
+        ):
+        """
+        Create the subset specification. If neither internal_subset_id nor a completed
+        subset_process are provided, the resultant instance is a null specification.
+        
+        internal_subset_id - the ID of the subset. If provided, subset_process is ignored.
+        subset_process - a process containing the ID of the subset. Ignored if internal_subset_id
+            is provided or the process is not complete.
```

Comment:
when would both `internal_subset_id` and `subset_process` be provided, and should we ensure that only one of them is provided? Can you also add a comment explaining why is internal_subset_id given precedence over subset_process?

Reply:
> when would both internal_subset_id and subset_process be provided

Never

> should we ensure that only one of them is provided? 

I don't really have a strong opinion because there's no reason for this to ever happen. I'm fine with throwing an error if both are provided if that's what you want.

> why is internal_subset_id given precedence over subset_process?

It's more or less arbitrary. Since I never expect both to be provided at the same time I didn't care too much about the interactions

Reply:
I would throw an error if both are provided.

Reply:
fixed